src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -115,7 +157,14 @@ static ParquetBufferValue parseColumnValueToParquet(
         throw new SFException(
             ErrorCode.INVALID_FORMAT_ROW, type.getName(), "Passed null to non nullable field");
       }
-      stats.incCurrentNullCount();
+      if (type.isPrimitive()) {
+        if (!statsMap.containsKey(path)) {
+          throw new SFException(
+              ErrorCode.INTERNAL_ERROR,
+              String.format("Stats not found for column: %s", type.getName()));
+        }
```

Comment:
move this if !containsKey to before the if (value != null) check instead of duplicating in both if (value == null) and if (value != null) code paths.

Reply:
Moved to the beginning.